pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -162,14 +162,19 @@ func (mi *StorageInitializerInjector) InjectStorageInitializer(pod *v1.Pod) erro
 		storageInitializerImage = mi.config.Image
 	}
 
+	modelLocalMountPath := constants.DefaultModelLocalMountPath
+	if modelName, ok := pod.ObjectMeta.Labels[constants.InferenceServicePodLabelKey]; ok {
+		modelLocalMountPath += "/" + modelName
+	}
+
 	securityContext := userContainer.SecurityContext.DeepCopy()
 	// Add an init container to run provisioning logic to the PodSpec
 	initContainer := &v1.Container{
 		Name:  StorageInitializerContainerName,
 		Image: storageInitializerImage,
 		Args: []string{
 			srcURI,
-			constants.DefaultModelLocalMountPath,
+			modelLocalMountPath,
```

Comment:
this is likely a non backwards compatible change as some user might depend on the old model path